scripts/generate.go

```diff
@@ -0,0 +1,374 @@
+	}
+
+	actionName := workflowDirParts[1]
+	actionPath := path.Join(workflowDirParts[:2]...)
+	actionReadMePath := path.Join(actionPath, "README.md")
+
+	if _, ok := wc[workflowID]; ok {
+		return fmt.Errorf("workflow exists in %s, please use existing workflow or use a different name", workflowConfigPath)
+	}
+
+	if _, err := os.Stat(workflowFilePath); err == nil {
+		return fmt.Errorf("workflow file %s already exists", workflowFilePath)
+	}
+
+	if err := os.MkdirAll(workflowDir, 0755); err != nil {
+		return fmt.Errorf("failed to create workflow directory: %w", err)
+	}
+
+	_, err := os.Stat(actionReadMePath)
+	if os.IsNotExist(err) {
+		actionReadMeContents := fmt.Sprintf("# %s examples", actionName)
+		if err := os.WriteFile(actionReadMePath, []byte(actionReadMeContents), 0644); err != nil {
+			return fmt.Errorf("failed writing content to action README file %s: %w", actionReadMePath, err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("failed to validate %s exists: %w", actionReadMePath, err)
+	}
+
+	fileContents := "# TODO: Add meaningful workflow content here."
+	if err := os.WriteFile(workflowFilePath, []byte(fileContents), 0644); err != nil {
+		return fmt.Errorf("writing content to workflow file: %w", err)
+	}
+
+	propertiesFilePath := path.Join(propertiesDirName, fmt.Sprintf("%s.properties.json", workflowID))
+	propertiesConfig := &propertiesTemplateConfig{
+		WorkflowID: workflowID,
+	}
+
+	if err := renderTemplate(propertiesTemplPath, propertiesFilePath, propertiesConfig); err != nil {
+		return fmt.Errorf("failed to render properties template: %w", err)
+	}
+
+	wc[workflowID] = workflow{
+		Starter:        *starterPtr,
+		Type:           *typePtr,
+		WorkflowPath:   workflowFilePath,
+		PropertiesPath: propertiesFilePath,
+	}
+
+	newConfigBytes, err := json.MarshalIndent(wc, "", "  ")
+	if err != nil {
+		return fmt.Errorf("fail to marshal new workflow config: %w", err)
+	}
+
+	if err := os.WriteFile(workflowConfigPath, newConfigBytes, 0644); err != nil {
+		return fmt.Errorf("failed to write update workflow config: %w", err)
+	}
+
+	return nil
+}
+
+// generateWorkflow handles the creationg of the main readme and individual action readmes
+func generateReadme(ctx context.Context) error {
+	var wfConfig workflowConfig
+	if err := loadJSONFromFile(&wfConfig, workflowConfigPath); err != nil {
+		return fmt.Errorf("failed to load workflow config %s: %w", workflowConfigPath, err)
+	}
+
+	hasInvalidConfigs := false
+	sortedWorkflowsIDs := getSortedWorkflowIDs(wfConfig)
+	readmeActions := map[string]readmeAction{}
+
+	for _, workflowID := range sortedWorkflowsIDs {
+		workflow := wfConfig[workflowID]
+		workflowPathParts := strings.Split(workflow.WorkflowPath, "/")
+
+		// This should be at least workflows/action-name/workflow-name.yml, but can be longer
+		if len(workflowPathParts) < 3 {
+			return fmt.Errorf("invalid workflow path %s", workflow.WorkflowPath)
```

Comment:
nit: so users know how to fix the error
```suggestion
			return fmt.Errorf("invalid workflow path %s, should be at least action-name/workflow-name", workflow.WorkflowPath)
```